utilities/WaxSession.php

```diff
@@ -7,16 +7,13 @@
     $response->add_header('P3P', ' CP="NOI ADM DEV PSAi COM NAV OUR OTRo STP IND DEM"');
     
     //garbage collection
-    $cmd = "php ".__FILE__." ";
-    $run_garbage_collection = true;
     foreach(WaxSession::$garbage_collection_folders as $dir){
-      if(($stats = stat($dir."/garbage.collect.lock")) && time() < $stats[9]){
-        $run_garbage_collection = false;
-        break;
+      $stats = stat("$dir/garbage.collect.lock");
+      if($stats && $stats[9] + WaxSession::$garbage_collection_timeout < time()){
+        touch("$dir/garbage.collect.lock");
+        exec("cd $dir && find . -mmin ".(WaxSession::$garbage_collection_timeout / 60)." -exec rm -f \"{}\" \\;"); //mmin is in minutes
```

Comment:
my plan was to run rm -rf based on an event coming from a render of a request coming in from a remote request. I feel ashamed and very glad it never got merged.

Also, hi Ross! ❤️